pkg/mutation/stats_reporter.go

```diff
@@ -0,0 +1,94 @@
+package mutation
+
+import (
+	"context"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/metrics"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+const (
+	mutationSystemIterationsMetricName = "mutation_system_iterations"
+)
+
+// SystemConvergenceStatus defines the outcomes of the attempted mutation of an object by the
+// mutation System.  The System is meant to converge on a fully mutated object.
+type SystemConvergenceStatus string
+
+var (
+	systemConvergenceKey = tag.MustNewKey("success")
+
+	// JULIAN need to try making this a boolean somehow.
+
+	// SystemConvergenceTrue denotes a successfully converged mutation system request.
+	SystemConvergenceTrue SystemConvergenceStatus = "true"
+	// SystemConvergenceFalse denotes an unsuccessfully converged mutation system request.
+	SystemConvergenceFalse SystemConvergenceStatus = "false"
+
+	systemIterationsM = stats.Int64(
+		mutationSystemIterationsMetricName,
+		"The distribution of Mutator ingestion durations",
+		stats.UnitDimensionless)
+)
+
+func init() {
+	if err := register(); err != nil {
+		panic(err)
+	}
+}
+
+// StatsReporter reports mutator-related metrics.
+type StatsReporter interface {
+	ReportIterationConvergence(scs SystemConvergenceStatus, iterations int) error
+}
+
+// reporter implements StatsReporter interface.
+type reporter struct {
+	ctx context.Context
+}
+
+// NewStatsReporter creaters a reporter for webhook metrics.
+func NewStatsReporter() (StatsReporter, error) {
+	ctx, err := tag.New(
```

Comment:
ditto above. Just use context.TODO().

Reply:
@sozercan Removing the use of `tag.New()` removes the possibility of creating an error from the instantiation of a `reporter`.  `tag.New()` can produce an error because it has the ability to mutate the key/value pairs stored inside of a `context.Context` type.  As we feed in an empty `context.Context` and add no new information, there is no way for an error to be produced.

This really has two implications:
1. `NewStatsReporter` is reduced to having essentially no logic.  Perhaps we should just remove it?  We could use the `reporter` type (renamed to `Reporter`) and still specify the `StatsReporter` interface in function signatures.
1. `reporter` no longer needs to contain a `context`.  That context is useless and is essentially cruft.  We could just as easily create a fresh context in each of the reporting functions.

Reply:
@maxsmythe